File: Assignment___33/Program2.c
//_______________________________________________________________________________________________________________________________
//
//  File name :     Program2.c
//
//  Description :   Write a program which accept two numbers from user and display position of common ON bits from that two numbers.
//
//                  Input :  10    15
//                  Output : 2      4
//
//  Author :        Sarika Devidas Gadhe
//
//_______________________________________________________________________________________________________________________________



#include<stdio.h>
#include<stdbool.h>

typedef unsigned int UNIT;

void CommonOnBits(UNIT iNo1,UNIT iNo2)
{
    UNIT iResult = iNo1 & iNo2;

    int iCount = 0;
    
    while(iResult != 0)
    {
        if(iResult & 1)
        {
            printf("%d\n",iCount);
        }
        iResult = iResult >> 1;
        iCount++;

    }
    printf("\n");

}

int main()
{
    UNIT iValue1 = 0,iValue2 = 0;

    printf("Enter the 1st number : ");
    scanf("%d",&iValue1);

    printf("Enter the 2nd number : ");
    scanf("%d",&iValue2);

    CommonOnBits(iValue1,iValue2);
   
    return 0;

}

File: Assignment___47/Program2.java
//_______________________________________________________________________________________________________________________________
//
//  File name :     Program2.java
//
//  Description :   Write program which accept number of rows and number of columns from user and display below pattern.
//                  
//                  Input :     iRow = 4        iCol = 3
//                  
//                  Output :    
//                              1   2   3      
//                              1   2   3     
//                              1   2   3     
//                              1   2   3       
//                                   
//
//  Author :        Sarika Devidas Gadhe
//
//_______________________________________________________________________________________________________________________________


import java.util.*;

class Pattern
{
    public void Pattern(int iRow , int iCol)
    {
        int i = 0, j = 0;

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                System.out.print(j+"\t");  
            }
            
            System.out.println();
        }
       
    }
}

class Program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter number of rows : ");
        iValue1 = sobj.nextInt();
        
        System.out.println("Enter number of columns : ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Pattern(iValue1,iValue2);

        sobj.close();


    }
}